Shell/5976

cosimoc opened the pull request:
endlessm/eos-shell#5976

cosimoc committed:
Adapt to avoid absolute paths

eos-shell-content will not put absolute paths in desktop files for
bundles that require custom splash screens anymore. Resolve relative
paths in eos-speedwagon.

[endlessm/eos-shell#5976]

cosimoc committed:
Split out two functions

For readability, split out logical functions.

[endlessm/eos-shell#5976]

dbnicholson commented on src/eos-speedwagon.c:
```diff
@@ -43,6 +43,78 @@ typedef struct {
+                                       SPLASH_BACKGROUND_DESKTOP_KEY);
+
+      /* rewrite absolute paths */
+      if (g_path_is_absolute (bg_name))
```
Instead of always rewriting absolute paths, maybe it would be a good idea to check first if it exists and return early if it does. I don't see us really needing that functionality, but it seems more robust if you're in a situation where speedwagon is updated but shell-content isn't. E.g.

```
if (g_path_is_absolute (bg_name))
  {
    if (g_file_test (bg_name, G_FILE_TEST_EXISTS))
      return g_strdup (bg_name);
    /* continue with basename logic */
  }
```

Although, I guess this is what @magcius was objecting to.

magcius replied:
I was thinking of that as well. I think checking whether it exists would
also fix my bundle issue, but I'd still rather just wait if given an option.

On Fri, Nov 13, 2015, 10:06 AM Dan Nicholson notifications@github.com
wrote:

> In src/eos-speedwagon.c
> https://github.com/endlessm/eos-speedwagon/pull/4#discussion_r44813462:
> 
> > @@ -43,6 +43,78 @@ typedef struct {
> >  GType eos_speedwagon_window_get_type (void) G_GNUC_CONST;
> >  G_DEFINE_TYPE (EosSpeedwagonWindow, eos_speedwagon_window, GTK_TYPE_APPLICATION_WINDOW)
> > 
> > +static gchar *
> > +get_splash_background_path (GDesktopAppInfo *app_info)
> > +{
> > -  gchar *bg_path = NULL;
> >   +
> > -  if (g_desktop_app_info_has_key (app_info,
> > -                                  SPLASH_BACKGROUND_DESKTOP_KEY))
> > -    {
> > -      gchar *bg_name =
> > -        g_desktop_app_info_get_string (app_info,
> > -                                       SPLASH_BACKGROUND_DESKTOP_KEY);
> >   +
> > -      /\* rewrite absolute paths */
> > -      if (g_path_is_absolute (bg_name))
> 
> Instead of always rewriting absolute paths, maybe it would be a good idea
> to check first if it exists and return early if it does. I don't see us
> really needing that functionality, but it seems more robust if you're in a
> situation where speedwagon is updated but shell-content isn't. E.g.
> 
> if (g_path_is_absolute (bg_name))
>   {
>     if (g_file_test (bg_name, G_FILE_TEST_EXISTS))
>       return g_strdup (bg_name);
>     /\* continue with basename logic */
>   }
> 
> Although, I guess this is what @magcius https://github.com/magcius was
> objecting to.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/endlessm/eos-speedwagon/pull/4/files#r44813462.

cosimoc replied:
OK I'll rework the patch to add that bit.

cosimoc committed:
Use absolute path if it exists

When the file at the absolute path specified by the splash screen key
exists, use it instead of rewriting it.

[endlessm/eos-shell#5976]

cosimoc commented:
@dbnicholson CR ready again

dbnicholson commented:
This looks good to me and I agree with @cosimoc that this is a step in the right direction. @magcius unless you object, I think we should merge.

dbnicholson commented:
I'm assuming that no comment from @magcius means agreement. Merging.

magcius commented:
My comment about wanting to wait until we ship splash screens in their own
bundles still holds -- I don't see any value in this at all in the short
term, but I'm not really going to fight it if you two want this.

On Fri, Nov 20, 2015, 10:03 AM Dan Nicholson notifications@github.com
wrote:

> I'm assuming that no comment from @magcius https://github.com/magcius
> means agreement. Merging.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/endlessm/eos-speedwagon/pull/4#issuecomment-158476893
> .
